Packs/QualysFIM/Integrations/Qualys/QualysFIM.py

```diff
@@ -0,0 +1,643 @@
+                                    'pageSize': args.get('limit', None),
+                                    'attributes': args.get('attributes', None)})
+    raw_response = client.get_incident_events(args.get('incident_id', None), params)
+    table_headers = ['id', 'name', 'severity', 'action', 'type', 'dateTime']
+    outputs = []
+    raw_outputs = []
+    if raw_response:
+        for item in raw_response:
+            data = item.get('data', None)
+            if data:
+                raw_outputs.append(data)
+                item_dictionary = create_event_or_incident_output(data, table_headers)
+                date_time = item_dictionary.get('dateTime', None)
+                if date_time:
+                    date_time = datetime.strptime(date_time,
+                                                  DATETIME_FORMAT).strftime(TABLE_DATETIME_FORMAT)
+                    item_dictionary['dateTime'] = date_time
+                outputs.append(item_dictionary)
+    readable_output = tableToMarkdown(name=f'Listed {len(outputs)} Events From Incident:',
+                                      t=outputs, headers=table_headers, removeNull=True)
+    return CommandResults(outputs_prefix='QualysFIM.Event',
+                          outputs_key_field='id',
+                          raw_response=raw_response,
+                          outputs=raw_outputs,
+                          readable_output=readable_output)
+
+
+def create_incident_command(client: Client, args: dict):
+    """
+    Create a new incident.
+
+    Args:
+        client (Client): Qualys FIM API client.
+        args (dict): All command arguments.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    if len(str(args.get('name', None))) > 128:
+        raise ValueError('Name is limited to 128 characters, please shorten name.')
+    if args.get('filters'):
+        filters = (args.get('filters'))
+    else:
+        if args.get('from_date') and args.get('to_date'):
+            filters = f"dateTime: ['{args.get('from_date')}'..'{args.get('to_date')}']"
+        else:
+            yesterday = (datetime.today() - timedelta(days=1)).strftime(DATETIME_FORMAT)
+            today = datetime.today().strftime(DATETIME_FORMAT)
+            filters = f"dateTime: ['{yesterday}'..'{today}']"
+    data = remove_empty_elements({'name': args.get('name', None), 'type': 'DEFAULT',
+                                  'filters': [f"{filters}"],
+                                  'reviewers': argToList(args.get('reviewers', None))})
+    raw_response = client.create_incident(data)
+    table_headers = ['id', 'name', 'reviewers', 'username', 'occurred', 'filters', 'approvalType']
+
+    output = create_event_or_incident_output(raw_response, table_headers)
+    output.update({'username': dict_safe_get(raw_response, ['userInfo', 'user', 'name'])})
+    output.update({'occurred': epochToTimestamp(dict_safe_get(raw_response, ['userInfo', 'date']))})
```

Comment:
Is there a possibility that the `dict_safe_get` will not return the correct value? If so we need to verify it before updating the output.

Reply:
no the response will always contain the creation date of the incident